src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -72,6 +72,24 @@ void CgroupV1Controller::set_subsystem_path(char *cgroup_path) {
             buf[MAXPATHLEN-1] = '\0';
             _path = os::strdup(buf);
           }
+        } else {
+          // Find the longest common substring as namespaces are hierarchical.
+          // If there are no matches between the root and the cgroup path,
+          // the controller's mount point will be used, which is a reasonable
+          // fallback.
+          stringStream ss;
+          ss.print_raw(_mount_point);
+          const char* root_p = _root;
+          const char* cgroup_p = cgroup_path;
+          int last_matching_slash_pos = -1;
+          for (int i = 0; *root_p == *cgroup_p && *root_p != 0; i++) {
+            if (*root_p == '/') {
+              last_matching_slash_pos = i;
+            }
+            root_p++; cgroup_p++;
+          }
+          ss.print_raw(_root, last_matching_slash_pos);
+          _path = os::strdup(ss.base());
```

Comment:
Do you mean `Find the longest common prefix`? Maybe give an example in the comments? Text parsing in C code is really difficult to understand.

Reply:
Maybe factor out the search like this
```
// Return length of common starting substring. E.g. "cat" for ("cattle", "catnip");
static int find_common_starting_substring(const char* s1, const char* s2) {
...
}
```
That way its clearer and we can find and move this to utilities if we ever need this in other places.

Reply:
OK.

Reply:
@iklam yes I meant `Find the longest common prefix`. Fixed the comment.

Reply:
I'm not convinced the extra function makes the code more readable, but here it is. I can revert back if this is too much.

Reply:
@jerboaa Feel free to go back to your original variant. This was only a proposal.

Reply:
understood.